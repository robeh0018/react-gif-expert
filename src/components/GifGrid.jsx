import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
    
    const {images, isLoading} = useFetchGifs(category);

    return ( 
        <>
            <h3> { category } </h3>
            {
                isLoading && (<h2>Loading...</h2>)
            }


            <div className="card-grid">
                { 
                    images.map(({id, title, url}) => (
                        <GifGridItem 
                            key={id}
                            title={title}
                            url={url}
                            // {...image}
                        />
                    ))
                } 
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
 