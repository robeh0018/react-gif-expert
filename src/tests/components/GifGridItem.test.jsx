import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../components";

const title = 'One puch man';
const url = 'http://localhost/asdfeqwfa';

describe('prueba a GifGridItem', () => { 
    
    test('should debe mostrar un title', () => { 
        render(<GifGridItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
     })
    
     test('should mostrar  una imagem con el url y el alt indicado', () => { 
        render(<GifGridItem title={title} url={url}/>)
         // expect(screen.getByRole('img', {src: {url}, alt: {title}})).toBeTruthy();
         // expect(screen.getByRole('img').src).toBe(url);
         // expect(screen.getByRole('img').alt).toBe(title);
         const {src, alt} = screen.getByRole('img');
         expect(src).toBe(url);
         expect(alt).toBe(title);
      });
    
     test('should hacer match con el snapshot', () => { 
        const {container} = render(<GifGridItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })
 })