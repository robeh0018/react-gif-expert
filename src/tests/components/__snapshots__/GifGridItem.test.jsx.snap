// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba a GifGridItem should hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One puch man"
      src="http://localhost/asdfeqwfa"
    />
    <p>
      One puch man
    </p>
  </div>
</div>
`;
